js/Application.js

```diff
@@ -221,19 +215,23 @@ define(
 						if (config.userAuthenticationEnabled && !authApi.isAuthenticated()) {
 							this.authSubscription = authApi.isAuthenticated.subscribe(async (isAuthed) => {
 								if (isAuthed) {
-									await sourceApi.initSourcesConfig();
+									sharedState.appInitializationStatus(await sourceApi.initSourcesConfig());
 									this.authSubscription.dispose();
 									console.info('Re-initialized service information');
 								}
 							});
-
+							sharedState.appInitializationStatus(constants.applicationStatuses.running);
 							resolve();
 							return;
 						} else {
-							sourceApi.initSourcesConfig().then(() => {
-								console.info('Init sources from server');
-								resolve();
-							});
+							sourceApi.initSourcesConfig()
+								.then(function (appStatus) {
+									sharedState.appInitializationStatus(appStatus);
+								})
+								.then(() => {
```

Comment:
Why do you need `then` here if previous step (`sharedState.appInitializationStatus(appStatus);`) doesn't return promise?

Reply:
all the next `.then` will trigger if previous `.then` returns undefined

Reply:
I haven't said that they won't. I was confused why the step was moved into separate `then` (earlier it wasn't)?